src/Tools/Program.cs

```diff
@@ -0,0 +1,337 @@
+                        case "HELP":
+                            return usage();
+
+                        default:
+                            var index = arg.IndexOf(':');
+                            if (index == -1 || index == arg.Length - 1)
+                            {
+                                return usage();
+                            }
+
+                            var key = arg.Substring(0, index).ToUpperInvariant();
+                            var value = arg.Substring(index + 1);
+                            switch (key)
+                            {
+                                case "P":
+                                case "PROJECT":
+                                    if (!File.Exists(value))
+                                    {
+                                        return fileNotExists(value);
+                                    }
+
+                                    if (!value.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase) &&
+                                        !value.EndsWith(".vbproj", StringComparison.OrdinalIgnoreCase))
+                                    {
+                                        return notASupportedProject(value);
+                                    }
+
+                                    projectsOrSolutions.Add(value);
+                                    break;
+
+                                case "S":
+                                case "SOLUTION":
+                                    if (!File.Exists(value))
+                                    {
+                                        return fileNotExists(value);
+                                    }
+
+                                    if (!value.EndsWith(".sln", StringComparison.OrdinalIgnoreCase))
+                                    {
+                                        return notASolution(value);
+                                    }
+
+                                    projectsOrSolutions.Add(value);
+                                    break;
+
+                                case "O":
+                                case "OUT":
+                                    if (value.Length == 0)
+                                    {
+                                        return invalidOutputFile(value);
+                                    }
+
+                                    outputFile = value;
+                                    break;
+
+                                default:
+                                    return usage();
+                            }
+
+                            break;
+                    }
+                }
+
+                if (projectsOrSolutions.Count == 0)
+                {
+                    return requiresProjectOrSolution();
+                }
+
+                return ErrorCode.None;
+            }
+
+            ErrorCode usage()
+            {
+                Console.WriteLine(@"
+Usage: Metrics.exe <arguments>
```

Comment:
localization?

Reply:
Tracked in https://github.com/dotnet/roslyn-analyzers/issues/1756